File: core/src/test/java/in/testpress/ui/BaseToolBarActivityTest.java
package in.testpress.ui;

import androidx.appcompat.app.AppCompatActivity;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.powermock.api.mockito.PowerMockito;
import org.powermock.core.classloader.annotations.PrepareForTest;
import org.powermock.modules.junit4.PowerMockRunner;

import in.testpress.network.RetrofitCall;
import in.testpress.util.CommonUtils;

import static org.junit.Assert.fail;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;
import static org.powermock.api.mockito.PowerMockito.doCallRealMethod;
import static org.powermock.api.mockito.PowerMockito.doNothing;
import static org.powermock.api.mockito.PowerMockito.when;
import static org.powermock.api.support.membermodification.MemberMatcher.methodsDeclaredIn;


@RunWith(PowerMockRunner.class)
@PrepareForTest({ AppCompatActivity.class, CommonUtils.class })
public class BaseToolBarActivityTest {

    @Mock
    private RetrofitCall retrofitCall;

    @Test
    public void test_onStop_cancelAPIRequests_isCalled() {
        PowerMockito.suppress(methodsDeclaredIn(AppCompatActivity.class));
        BaseToolBarActivity activity = mock(BaseToolBarActivity.class);

        RetrofitCall[] retrofitCalls = new RetrofitCall[] { retrofitCall };

        when(activity.getRetrofitCalls()).thenReturn(retrofitCalls);

        doCallRealMethod().when(activity).onDestroy();
        activity.onDestroy();

        verify(activity, times(1)).getRetrofitCalls();
        verify(retrofitCall, times(1)).cancel();

        PowerMockito.mockStatic(CommonUtils.class);
        try {
            doNothing().when(CommonUtils.class, "cancelAPIRequests", (Object) retrofitCalls);
        } catch (Exception e) {
            e.printStackTrace();
            fail();
        }

        activity.onDestroy();

        PowerMockito.verifyStatic(CommonUtils.class, times(1));
        CommonUtils.cancelAPIRequests(retrofitCalls);
    }
}

File: core/src/test/java/in/testpress/ui/BaseToolBarActivityTest.kt
package `in`.testpress.ui

import `in`.testpress.network.RetrofitCall
import `in`.testpress.util.CommonUtils
import androidx.appcompat.app.AppCompatActivity
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.Mockito
import org.powermock.api.mockito.PowerMockito
import org.powermock.api.support.membermodification.MemberMatcher
import org.powermock.core.classloader.annotations.PrepareForTest
import org.powermock.modules.junit4.PowerMockRunner

@RunWith(PowerMockRunner::class)
@PrepareForTest(AppCompatActivity::class, CommonUtils::class)
class BaseToolBarActivityTest {
    @Mock
    private lateinit var retrofitCall: RetrofitCall<*>
    @Test
    fun test_onStop_cancelAPIRequests_isCalled() {
        retrofitCall = Mockito.mock(RetrofitCall::class.java)
        PowerMockito.suppress(MemberMatcher.methodsDeclaredIn(AppCompatActivity::class.java))
        val activity = Mockito.mock(
            BaseToolBarActivity::class.java
        )
        val retrofitCalls: Array<RetrofitCall<*>> = arrayOf(retrofitCall)
        PowerMockito.`when`(activity.getRetrofitCalls())
            .thenReturn(retrofitCalls)
        PowerMockito.doCallRealMethod().`when`(activity).onDestroy()
        activity.onDestroy()
        Mockito.verify(activity, Mockito.times(1)).getRetrofitCalls()
        Mockito.verify(retrofitCall, Mockito.times(1)).cancel()
        PowerMockito.mockStatic(CommonUtils::class.java)
        try {
            PowerMockito.doNothing().`when`(
                CommonUtils::class.java,
                "cancelAPIRequests",
                retrofitCalls as Any
            )
        } catch (e: Exception) {
            e.printStackTrace()
            Assert.fail()
        }
        activity.onDestroy()
        PowerMockito.verifyStatic(CommonUtils::class.java, Mockito.times(1))
        CommonUtils.cancelAPIRequests(retrofitCalls)
    }
}